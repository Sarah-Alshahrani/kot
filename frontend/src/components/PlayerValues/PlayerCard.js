import React from "react";
import Info from "./../Images/icons8-info-128.png";
import Energy from "./../Images/Energy.png";
import Footnote from "./../Images/icons8-footsteps-50.png";
import "./PlayerCard.css";

class PlayerCard extends React.Component {
  render() {
    let t_footnote = "None";
    if (this.props.footnote) {
      t_footnote = this.props.footnote;
    }

    return (
      <div>
        <div className="card-border">
          <p className="card-instances">
            {this.props.type}
            {this.props.name}
            <br />
            <span>
              {this.props.cost}
              <img className="img-energy" alt="" src={Energy}></img>
            </span>
            <span data-tooltip={this.props.effect}>
              <img className="img-info" alt="" src={Info}></img>
            </span>
            <span data-tooltip={t_footnote}>
              <img className="img-info" alt="" src={Footnote}></img>
            </span>
          </p>
        </div>
      </div>
    );
  }
}
export default PlayerCard;
