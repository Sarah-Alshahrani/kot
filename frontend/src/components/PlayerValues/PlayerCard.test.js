import React from "react";
import { shallow } from "enzyme";

import PlayerCard from "./PlayerCard";

describe("Verify card detail template", () => {
  it("Card details should be rendered correctly", () => {
    const errorPage = shallow(<PlayerCard />);
    expect(errorPage).toMatchSnapshot();
  });
});
