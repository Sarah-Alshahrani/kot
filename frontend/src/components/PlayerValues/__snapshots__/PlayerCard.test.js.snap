// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify card detail template Card details should be rendered correctly 1`] = `
<div>
  <div
    className="card-border"
  >
    <p
      className="card-instances"
    >
      <br />
      <span>
        <img
          alt=""
          className="img-energy"
          src="Energy.png"
        />
      </span>
      <span>
        <img
          alt=""
          className="img-info"
          src="icons8-info-128.png"
        />
      </span>
      <span
        data-tooltip="None"
      >
        <img
          alt=""
          className="img-info"
          src="icons8-footsteps-50.png"
        />
      </span>
    </p>
  </div>
</div>
`;
